File: dmr-app/src/app/_core/_model/glue-type.ts
export class HierarchyNode<T> {
    childNodes: Array<HierarchyNode<T>>;
    depth: number;
    hasChildren: boolean;
    parent: T;
    entity: T;
    constructor() {
        this.childNodes = new Array<HierarchyNode<T>>();
    }
    any(): boolean {
        return this.childNodes.length > 0;
    }
}
export interface IGlueType {
    id: number;
    level: number;
    title: string;
    method: string;
    parentID: number;
    rpm: number;
    minutes: number;
}

File: dmr-api/Models/GlueType.cs
namespace dmr_api.Models
{
    public class GlueType
    {
        public int ID { get; set; }
        public string Code { get; set; }
        public string Title { get; set; }
        public int Level { get; set; }
        public double Minutes { get; set; }
        public double RPM { get; set; }
        public string Method { get; set; }

        public int? ParentID { get; set; }
    }
}